app/services/schedules/find.rb

```diff
@@ -0,0 +1,94 @@
+module Schedules
+  class Find
+    attr_accessor :period, :training_programme, :started_on
+
+    def initialize(period:, training_programme:, started_on:)
+      @period = period
+      @training_programme = training_programme
+      @started_on = started_on
+    end
+
+    def call
+      return unless provider_led?
+      return most_recent_schedule if previous_provider_led_periods.exists?
+
+      Schedule.find_by(contract_period_year:, identifier:)
+    end
+
+  private
+
+    def provider_led?
+      training_programme == "provider_led"
+    end
+
+    def previous_provider_led_periods
+      period.teacher.training_periods.where(training_programme: 'provider_led')
+    end
+
+    def most_recent_provider_led_period
+      previous_provider_led_periods.latest_first.first
+    end
```

Comment:
I'm just wondering with this set of private methods - would they be better centralised? I would be surprised if they didn't already exist elsewhere but if they don't (and I did a quick grep and couldn't find them...) they seem like ones that we might end up wanting to use elsewhere and so might end up defining in multiple places